src/cockpit/channels/http_stream2.py

```diff
@@ -0,0 +1,130 @@
+# This file is part of Cockpit.
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import http.client
+
+from ..channel import Channel
+
+logger = logging.getLogger(__name__)
+
+
+class HttpStream2Channel(Channel):
+    payload = 'http-stream2'
+    connections = {}
+
+    def get_connection(self, name):
+        if name and name in HttpStream2Channel.connections:
```

Comment:
In general, this "check and then access" pattern is typically considered unpythonic.

Either (my preference) use `.get()` (possibly with a walrus) or just do the lookup directly and catch the exception.

Reply:
Used try/except KeyError.

Reply:
Remember, no walruses yet -- RHEL 8!